### PR TITLE
Add Glitchtip support

### PR DESCRIPTION
**Pull request by tpapaioa**
# Description
This PR implements GlitchTip for error tracking:
- add new `GlitchtipDsn` configuration parameter, which is read as a Vault secret when deployed via Clowder
- add new dependency `github.com/getsentry/sentry-go`
- initialize the Sentry client in `main.go`, so that all exceptions caught by the client in `main()` will be sent to GlitchTip

Normal functionality is unaffected, and I've verified that a `panic()` added after https://github.com/RedHatInsights/edge-api/blob/cc3990138b72f90ca592cf893b8b72770be3ba4c/main.go#L179 is successfully reported to the edge-api-stage GlitchTip project for PR checks run against stage.

FIXES: THEEDGE-2974

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -167,6 +168,16 @@ func main() {
 	initDependencies()
 
 	cfg := config.Get()
+
+	// Set up Sentry client for GlitchTip error tracking
+	sentry.Init(sentry.ClientOptions{
```
Comment (omaciel) [inline main.go:173]: Perhaps we could create a FeatureFlag for this so that it's easier to test, enable/disable? Otherwise, looks good

Comment (de1987) [reply main.go:173]: Agree

Comment (tpapaioa) [reply main.go]: @omaciel @de1987 I've added a new feature flag (currently toggled off) in Unleash and added a check before enabling the client.

File: unleash/features/feature.go [hunk 2]
```diff
@@ -23,6 +23,11 @@ type Flag struct {
 	EnvVar string
 }
 
+// GLITCHTIP LOGGING FLAGS
+
+// GlitchtipLogging is the feature flag for reporting errors to GlitchTip
+var GlitchtipLogging = &Flag{Name: "edge-management.glitchtip_logging", EnvVar: "GLITCHTIP_LOGGING"}
```
Comment (omaciel) [inline unleash/features/feature.go:29]: Where is GLITCHTIP_LOGGING defined?

Comment (tpapaioa) [reply unleash/features/feature.go:29]: Environment parameter overrides for feature flags are checked here: 

https://github.com/RedHatInsights/edge-api/blob/1604ec8ca789f1dc52bd2b83bfeddd75b35800a9/unleash/features/feature.go#L101

**Review by de1987 — APPROVED** [review]

**Review by omaciel — APPROVED** [review]
/lgtm

**Pull request merged**